haystack/components/generators/hugging_face_local.py

```diff
@@ -195,19 +194,71 @@ def to_dict(self) -> Dict[str, Any]:
         """
         Serialize this component to a dictionary.
         """
-        pipeline_kwargs_to_serialize = deepcopy(self.huggingface_pipeline_kwargs)
-
-        # we don't want to serialize valid tokens
-        if isinstance(pipeline_kwargs_to_serialize["token"], str):
-            pipeline_kwargs_to_serialize["token"] = None
-
-        return default_to_dict(
+        serialization_dict = default_to_dict(
             self,
-            huggingface_pipeline_kwargs=pipeline_kwargs_to_serialize,
+            huggingface_pipeline_kwargs=self.huggingface_pipeline_kwargs,
             generation_kwargs=self.generation_kwargs,
             stop_words=self.stop_words,
         )
 
+        huggingface_pipeline_kwargs = serialization_dict["init_parameters"]["huggingface_pipeline_kwargs"]
+        # we don't want to serialize valid tokens
+        if isinstance(huggingface_pipeline_kwargs["token"], str):
+            serialization_dict["init_parameters"]["huggingface_pipeline_kwargs"]["token"] = None
```

Comment:
For consistency I would `pop()` the item. Setting it to `None` might affect the behaviour of the code downstream

Reply:
done!